packages/radix-vue/src/Accordion/AccordionRoot.vue

```diff
@@ -1,34 +1,214 @@
+   *
+   * @default "ltr"
+   */
+  dir?: Direction;
+
+  /**
+   * The orientation of the accordion.
+   *
+   * @default "vertical"
+   */
+  orientation?: DataOrientation;
+};
+
+export const ACCORDION_INJECTION_KEY =
+  Symbol() as InjectionKey<AccordionProvideValue>;
+
+export type AccordionProvideValue = {
+  disabled?: AccordionRootProps["disabled"];
+  direction: AccordionRootProps["dir"];
+  orientation: AccordionRootProps["orientation"];
+  parentElement: Ref<HTMLElement | undefined>;
+  handleItemState(value: string, open: boolean): void;
+  isSingle: ComputedRef<boolean>;
+  modelValue: Ref<string | undefined | string[]>;
+  collapsible: boolean;
+};
+
+function validateModelValue({ type, modelValue }: AccordionRootProps) {
+  if (type === AccordionType.Single) {
+    if (Array.isArray(modelValue)) {
+      console.error(`Invalid prop \`value\` of value \`${modelValue}\` supplied to \`AccordionRoot\`, which type is \`single\`. The \`value\` prop must be:
+  - a string
+  - \`undefined\`
+
+If you want to use multiple values, use the \`AccordionRoot\` with type \`multiple\`.
+
+Defaulting to \`undefined\`.`);
+      return undefined;
+    }
+  } else if (type === AccordionType.Multiple) {
+    if (typeof modelValue === "string" || typeof modelValue === "undefined") {
+      console.error(`Invalid prop \`value\` of value \`${modelValue}\` supplied to \`AccordionRoot\`, which type is \`multiple\`. The \`value\` prop must be:
+- an array of strings
+- empty array (\`[]\`)
+
+If you want to use just one value, use the \`AccordionRoot\` with type \`single\`.
+
+Defaulting to empty array (\`[]\`).`);
+      return [];
+    }
+  }
+  console.log({ modelValue });
+  return modelValue;
+}
+
+function validateDefaultValue(props: AccordionRootProps) {
+  if (props.type === AccordionType.Multiple) {
+    if (Array.isArray(props.defaultValue)) {
+      console.log("array", props.defaultValue);
+      return props.defaultValue;
+    } else if (props.defaultValue === undefined) {
+      return [];
+    } else {
+      console.error(
+        `Invalid prop \`defaultValue\` of value \`${props.defaultValue}\` supplied to \`AccordionRoot\`, which type is \`multiple\`. The \`defaultValue\` prop must be:
+  - an array of strings
+  - empty array (\`[]\`)
+
+If you want to use just one value, use the \`AccordionRoot\` with type \`single\`.
+
+Defaulting to empty array (\`[]\`).`
+      );
+      return [];
+    }
+  } else if (props.type === AccordionType.Single) {
+    if (typeof props.defaultValue === "string") {
+      return props.defaultValue;
+    } else if (props.defaultValue === undefined) {
+      return undefined;
+    } else {
+      console.error(
+        `Invalid prop \`defaultValue\` of value \`${props.defaultValue}\` supplied to \`AccordionRoot\`, which type is \`single\`. The \`defaultValue\` prop must be:
+  - a string
+  - \`undefined\`
+
+Defaulting to \`undefined\`.`
+      );
+      return undefined;
+    }
+  }
+}
```

Comment:
But, do you want to keep the `validate` functions?

Reply:
> But, do you want to keep the `validate` functions?

yeah validate is ok. look how small togglegroup is, try to follow that without losing any functionality/maintainability

Reply:
> An idea: what about a composable like this:
> 
> [39a27fc](https://github.com/radix-vue/radix-vue/commit/39a27fce5ce35fa387d82aba2cfe943f58424a3f)

idk how it is useful, show how it can be used accross components

Reply:
With:

```ts
const { modelValue, changeModelValue } = useSingleOrMultipleValue(props, emits);
```

It is basically the code that we have in[ `ToggleGroupRoot`](https://github.com/onmax/radix-vue/blob/39a27fce5ce35fa387d82aba2cfe943f58424a3f/packages/radix-vue/src/ToggleGroup/ToggleGroupRoot.vue#L67)


<details><summary>And doing a quick search in VSCode, we have like 8</summary>
<img src="https://github.com/radix-vue/radix-vue/assets/22072217/2e567750-2826-49ee-bcf8-7580119192c8" />
</details>

Reply:
So we don't repeat the same validations and the same `changeValueModel` logic

Reply:
I think this composable is good. It handles logic for single/multiple for `modelValue`, and with validation 👍🏻.